lib/perl/Genome/Disk/Detail/Allocation/Copier.pm

```diff
@@ -0,0 +1,168 @@
+package Genome::Disk::Detail::Allocation::Copier;
+
+use strict;
+use warnings;
+
+use Genome;
+
+use Carp qw(confess);
+
+class Genome::Disk::Detail::Allocation::Copier {
+    is => 'Genome::Disk::Detail::StrictObject',
+
+    has => [
+        allocation_id => {
+            is => 'Text',
+            len => 64,
+        },
+        output_dir => {
+            is => 'Text',
+            len => 255,
+        },
+    ],
+};
+
+sub copy {
+    my $self = shift;
+
+    my $id = $self->allocation_id;
+
+    my $output_dir = $self->output_dir;
+
+    my ($allocation_object, $allocation_lock) =
+        Genome::Disk::Allocation->get_with_lock($id);
+    my $original_absolute_path = $allocation_object->absolute_path;
+    
+    # make shadow allocation
+    my %creation_params = $self->_get_copy_shadow_params($allocation_object);
+
+    # The shadow allocation is just a way of keeping track of our temporary
+    # additional disk usage during the copy.
+    my $shadow_allocation = Genome::Disk::Allocation->shadow_get_or_create(%creation_params);
+    my $shadow_absolute_path = $shadow_allocation->absolute_path;
+    my %rsync_params = (
+        source_directory => $original_absolute_path,
+        target_directory => $output_dir,
+    );
+    if ($allocation_object->is_archived) {
+        Genome::Sys->create_directory($shadow_absolute_path);
+        eval {
+            my $tar_path = $allocation_object->tar_path;
+            my $cmd = "tar -C $shadow_absolute_path -xf $tar_path";
+
+            # It's very possible that if no commit is on, the volumes/allocations
+            # being dealt with are test objects that don't exist out of this local
+            # UR context, so bsubbing jobs would fail.
+            if ($ENV{UR_DBI_NO_COMMIT}) {
+                Genome::Sys->shellcmd(cmd => $cmd);
+            }
+            else {
+                # If this process should be killed, the LSF job needs to be cleaned up
+                my ($job_id, $status);
+            
+                # Signal handlers are added like this so an anonymous sub can be
+                # used, which handles variables defined in an outer scope
+                # differently than named subs (in this case, $job_id,
+                # $allocation_object, and $archive_path).
+                my @signals = qw/ INT TERM /;
+                for my $signal (@signals) {
+                    $SIG{$signal} = sub {
+                        print STDERR "Cleanup activated within allocation, ",
+                            "cleaning up LSF jobs\n";
+                        eval { Genome::Sys->kill_lsf_job($job_id) } if $job_id;
+                        die "Received signal, exiting.";
+                    }
+                }
```

Comment:
It all started in Archiver.  It appears it was copy/paste to Unarchiver.  The Copier version is slightly different because there is no need to cleanup the archive directory.  That would be bad.

Now that you point out Unarchiver, it appears an unsuccessful unarchive will cleanup the archive_path as well.  That seems bad.